components/permissions/brave_permission_manager.cc

```diff
@@ -31,56 +32,35 @@ GURL BravePermissionManager::GetCanonicalOrigin(
                                                embedding_origin);
 }
 
-void BravePermissionManager::ResetPermissionViaContentSetting(
-    ContentSettingsType type,
-    const GURL& requesting_origin,
-    const GURL& embedding_origin) {
-  DCHECK_CURRENTLY_ON(content::BrowserThread::UI);
-  PermissionContextBase* context = GetPermissionContext(type);
-  if (!context)
-    return;
-  context->ResetPermission(
-      GetCanonicalOrigin(type, requesting_origin, embedding_origin),
-      url::Origin::Create(embedding_origin).GetURL());
-}
-
-void BravePermissionManager::RequestPermissionsDeprecated(
+void BravePermissionManager::RequestPermissionsForOrigin(
     const std::vector<blink::PermissionType>& permissions,
     content::RenderFrameHost* render_frame_host,
     const GURL& requesting_origin,
     bool user_gesture,
     base::OnceCallback<void(const std::vector<blink::mojom::PermissionStatus>&)>
         callback) {
-  RequestPermissions(permissions, render_frame_host, requesting_origin,
-                     user_gesture, std::move(callback));
+  DCHECK_CURRENTLY_ON(content::BrowserThread::UI);
+  base::AutoReset<GURL> auto_reset_requesting_origin(&forced_requesting_origin_,
```

Comment:
why do we still need requesting origin here? Can't we just pass the correct rfh?

Reply:
@goodov Do you think you could weigh in this question?

Reply:
I need to checkout this branch so I can see what's going on here more easily

Reply:
Wallet permission request bakes wallet address into the origin. There's no any other way to pass this data during the permission request.

Reply:
discussed some options in DM, we'll follow up